tools/js2c.cc

```diff
@@ -381,18 +360,47 @@ std::string GetVariableName(const std::string& id) {
   return result;
 }
 
-std::vector<std::string> GetCodeTable() {
-  size_t size = 1 << 16;
-  std::vector<std::string> code_table(size);
-  for (size_t i = 0; i < size; ++i) {
-    code_table[i] = std::to_string(i) + ',';
+
+
+// 382106 is the length of the string "0,1,2,3,...,65535,".
+// 65537 is 2**16 + 1
+constexpr std::pair<std::array<char, 382106>,std::array<uint32_t, 65537>> precompute_string() {
```

Comment:
this can be consteval as well, so we can ensure this runs on compile time. cc @lemire

Reply:
> this can be consteval as well, so we can ensure this runs on compile time.

It is designed to be consteval, but if we mark it as consteval, then we may get a build error if the compiler finds that the function is too expensive.

See this link: 

https://godbolt.org/z/o3q11Gz6z


GCC is happy to evaluate it at compile time but LLVM and Visual Studio do not want to.

We could convince LLVM to do it with compiler flags. I suspect that the same might be true of Visual Studio.

Reply:
Basically, there is a hidden limit set by the compiler to how much compile time work a function might do. It is usually never a problem in my experience, but here is a special case where we are building a gigantic table.